lib/util/viewer.js

```diff
@@ -464,3 +465,30 @@ export function generateFakeLegForRouteRenderer(
     routeTextColor: route.textColor
   }
 }
+
+/**
+ * Gets a breadcrumbs-like page title for the route and pattern viewer,
+ * so we don't need to internationalize the title bar structure:
+ * - if no route is selected (or pattern is loading), just say "Route Viewer",
+ * - if a route is selected, display the agency and user-facing route number,
+ * - if a pattern is active, also show the pattern destination.
+ */
+export function getRouteOrPatternViewerTitle(props) {
+  const { intl, transitOperators, viewedRoute, viewedRouteObject } = props
+  const { patternId } = viewedRoute || {}
+  const { patterns, pending, shortName } = viewedRouteObject || {}
+
+  if (!viewedRouteObject || pending) {
```

Comment:
It's clean to combine the two title functions! Maybe `routeIsLoading` instead of `pending`?